packages/twenty-front/src/modules/object-record/utils/getAvailableFieldsForAggregationFromObjectFields.ts

```diff
@@ -0,0 +1,67 @@
+import { FieldMetadataItem } from '@/object-metadata/types/FieldMetadataItem';
+import { AGGREGATE_OPERATIONS } from '@/object-record/record-table/constants/AggregateOperations';
+import { FieldMetadataType } from '~/generated-metadata/graphql';
+
+type AGGREGATE_OPERATIONS_OMITTING_COUNT = Exclude<
+  AGGREGATE_OPERATIONS,
+  AGGREGATE_OPERATIONS.count
+>;
+export type AvailableFieldsForAggregation = {
+  [T in AGGREGATE_OPERATIONS_OMITTING_COUNT]?: string[];
+};
+
+const FIELDS_AVAILABLE_BY_AGGREGATE_OPERATION = {
+  [AGGREGATE_OPERATIONS.min]: [
+    FieldMetadataType.Number,
+    FieldMetadataType.Currency,
+  ],
+  [AGGREGATE_OPERATIONS.max]: [
+    FieldMetadataType.Number,
+    FieldMetadataType.Currency,
+  ],
+  [AGGREGATE_OPERATIONS.avg]: [
+    FieldMetadataType.Number,
+    FieldMetadataType.Currency,
+  ],
+  [AGGREGATE_OPERATIONS.sum]: [
+    FieldMetadataType.Number,
+    FieldMetadataType.Currency,
+  ],
+};
+
+const initializeAggregationMap = (): AvailableFieldsForAggregation => {
+  return Object.keys(FIELDS_AVAILABLE_BY_AGGREGATE_OPERATION).reduce(
+    (acc, operation) => ({
+      ...acc,
+      [operation]: [],
+    }),
+    {},
+  );
+};
+
+const isFieldTypeValidForOperation = (
```

Comment:
We should keep one util / type per file, like we do in the codebase, it helps getting only what we need to read under our eyes faster and not scrolling in a file to look for the precise util we want to read or modify.

Reply:
I see, but I am not 100% confident this method here should belong to utils folder: it will only be called in this other util file. Still moving it out to follow your advice